src/net.cpp

```diff
@@ -725,12 +723,29 @@ int CNetMessage::readHeader(const char *pch, unsigned int nBytes)
         hdrbuf >> hdr;
     }
     catch (const std::exception&) {
+        LogPrint("net", "Header error: failed to deserialize");
         return -1;
     }
 
-    // reject messages larger than MAX_SERIALIZED_COMPACT_SIZE
-    if (hdr.nMessageSize > MAX_SERIALIZED_COMPACT_SIZE)
-            return -1;
+    // reject messages with invalid start
+    if (memcmp(hdr.pchMessageStart, Params().MessageStart(), MESSAGE_START_SIZE) != 0) {
+        LogPrint("net", "Header error: invalid start (%s)", SanitizeString(std::string(hdr.pchMessageStart, hdr.pchMessageStart + MESSAGE_START_SIZE)));
+        return -1;
+    }
+
+    // reject messages with invalid command
+    if (!hdr.HasValidCommand())
+    {        
+        LogPrint("net", "Header error: invalid command (%s)\n", SanitizeString(hdr.GetCommand()));
+        return -1;
+    }
+
+    // reject messages larger than MAX_SERIALIZED_COMPACT_SIZE or MAX_PROTOCOL_MESSAGE_LENGTH
+    if (hdr.nMessageSize > MAX_SERIALIZED_COMPACT_SIZE || hdr.nMessageSize > MAX_PROTOCOL_MESSAGE_LENGTH)
```

Comment:
Just nitpicking: considering MAX_PROTOCOL_MESSAGE_LENGTH is waaay lower than MAX_SERIALIZED_COMPACT_SIZE maybe reversing the order of checks would make sense.

Reply:
Done